### PR TITLE
pass exception to restart message

### PR DESCRIPTION
**Pull request by tomliversidge**
Added reason why a restart was deemed necessary

https://github.com/AsynkronIT/protoactor-dotnet/issues/228

### CODE REVIEW

**Comment by adamhathcock** [comment]
Are Reasons always going to be exceptions?

**Comment by tomliversidge** [comment]
I _think_ so. As far as I understand, actors only get restarted when their supervisors tell them to, and this supervision process only kicks in when an exception is thrown.

**Comment by tomliversidge** [comment]
```
var t = _invoker.InvokeUserMessageAsync(msg);
                        if (t.IsFaulted)
                        {
                            _invoker.EscalateFailure(t.Exception, msg);
                            continue;
                        }
```

is the code in the mailbox, so it's only on a faulted task that failure is escalted

**Comment by adamhathcock** [comment]
Works for me.  I was only thinking if someone wanted to have an explicit reason but that's probably an edge case that isn't worth accounting for.

File: src/Proto.Actor/Supervision.cs [hunk 1]
```diff
@@ -23,8 +23,8 @@ public enum SupervisorDirective
     public interface ISupervisor
     {
         IReadOnlyCollection<PID> Children { get; }
-        void EscalateFailure(PID who, Exception reason);
-        void RestartChildren(params PID[] pids);
+        void EscalateFailure(Exception reason, PID who);
```
Comment (tomliversidge) [inline src/Proto.Actor/Supervision.cs]: I switched this around to keep it consistent with RestartChildren, as that method uses params, which need to come last

**Pull request merged**